src/minecraft/kotlin/xyz/wagyourtail/unimined/internal/minecraft/patch/reindev/NoTransformReIndevTransformer.kt

```diff
@@ -0,0 +1,139 @@
+package xyz.wagyourtail.unimined.internal.minecraft.patch.reindev
+
+import org.gradle.api.Project
+import org.gradle.internal.cc.base.logger
+import xyz.wagyourtail.unimined.api.mapping.MappingNamespaceTree
+import xyz.wagyourtail.unimined.api.minecraft.EnvType
+import xyz.wagyourtail.unimined.api.minecraft.MinecraftJar
+import xyz.wagyourtail.unimined.api.minecraft.patch.reindev.ReIndevPatcher
+import xyz.wagyourtail.unimined.api.unimined
+import xyz.wagyourtail.unimined.internal.minecraft.MinecraftProvider
+import xyz.wagyourtail.unimined.internal.minecraft.patch.AbstractMinecraftTransformer
+import xyz.wagyourtail.unimined.util.FinalizeOnRead
+import xyz.wagyourtail.unimined.util.MustSet
+import xyz.wagyourtail.unimined.util.consumerApply
+import xyz.wagyourtail.unimined.util.stream
+import java.net.URI
+import java.nio.file.Path
+import kotlin.io.path.createDirectories
+import kotlin.io.path.exists
+import kotlin.io.path.outputStream
+import kotlin.math.pow
+
+class NoTransformReIndevTransformer(project: Project, provider: MinecraftProvider): AbstractMinecraftTransformer(
+    project,
+    provider,
+    "reindev"
+), ReIndevPatcher {
+
+    /**
+     * The ReIndev version to use
+     */
+    var version: String by FinalizeOnRead(MustSet())
+
+    val fileBaseURL: URI = URI.create("https://cdn.fox2code.com/files/")
+
+    val rindVersionFolder: Path by lazy {
+        project.unimined.getGlobalCache()
+            .resolve("net")
+            .resolve("silveros")
+            .resolve("reindev")
+            .resolve(version)
+    }
+
+    init {
+        addMavens()
+    }
+
+    fun addMavens() {
+        project.unimined.fox2codeMaven()
+    }
+
+    override var prodNamespace: MappingNamespaceTree.Namespace = provider.mappings.OFFICIAL
+
+    override fun transform(minecraft: MinecraftJar): MinecraftJar {
+        return minecraft.let(consumerApply {
+            val target = if (minecraft.envType == EnvType.CLIENT) reIndevClient else reIndevServer
+            if (target.path.exists() && !project.unimined.forceReload) {
+                return@consumerApply target
+            }
+
+            target
+        })
+    }
+
+    /**
+     * Non-metadata mirror of Utils.cachingDownload
+     */
+    private fun cachingDownload(
```

Comment:
How can I avoid having to use this function?

All I have available are these files: http://cdn.fox2code.com/files/

Reply:
you can use Utils.cachingDownload without metadata, it has default values for the metadata if it's not present, then it'll just act as a 1-day cache.

I use it like that in several places, such as
```kt
project.cachingDownload(
    URI.create("https://hub.spigotmc.org/versions/${rev}.json"),
    cachePath = buildInfoFile
)
```